core/src/main/java/com/scalar/db/common/CheckedDistributedStorageAdmin.java

```diff
@@ -0,0 +1,277 @@
+   * Whether to check if the namespace exists or not. Set false when the storage does not support
+   * namespaces.
+   */
+  private final boolean checkNamespace;
+
+  public CheckedDistributedStorageAdmin(DistributedStorageAdmin admin) {
+    this(admin, true);
+  }
+
+  public CheckedDistributedStorageAdmin(DistributedStorageAdmin admin, boolean checkNamespace) {
+    this.admin = admin;
+    this.checkNamespace = checkNamespace;
+  }
+
+  @Override
+  public void createNamespace(String namespace, Map<String, String> options)
+      throws ExecutionException {
+    if (checkNamespace && namespaceExists(namespace)) {
+      throw new IllegalArgumentException("Namespace already exists: " + namespace);
+    }
+
+    try {
+      admin.createNamespace(namespace, options);
+    } catch (ExecutionException e) {
+      throw new ExecutionException("Creating the namespace failed: " + namespace, e);
+    }
+  }
+
+  @Override
+  public void createTable(
+      String namespace, String table, TableMetadata metadata, Map<String, String> options)
+      throws ExecutionException {
+    if (checkNamespace && !namespaceExists(namespace)) {
+      throw new IllegalArgumentException("Namespace does not exist: " + namespace);
+    }
+    if (tableExists(namespace, table)) {
+      throw new IllegalArgumentException("Table already exists: " + namespace + "." + table);
+    }
+
+    try {
+      admin.createTable(namespace, table, metadata, options);
+    } catch (ExecutionException e) {
+      throw new ExecutionException(
+          "Creating the table failed: " + ScalarDbUtils.getFullTableName(namespace, table), e);
+    }
+  }
+
+  @Override
+  public void dropTable(String namespace, String table) throws ExecutionException {
+    if (!tableExists(namespace, table)) {
+      throw new IllegalArgumentException("Table does not exist: " + namespace + "." + table);
+    }
+
+    try {
+      admin.dropTable(namespace, table);
+    } catch (ExecutionException e) {
+      throw new ExecutionException(
+          "Dropping the table failed: " + ScalarDbUtils.getFullTableName(namespace, table), e);
+    }
+  }
+
+  @Override
+  public void dropNamespace(String namespace) throws ExecutionException {
+    if (checkNamespace && !namespaceExists(namespace)) {
+      throw new IllegalArgumentException("Namespace does not exist: " + namespace);
+    }
+    if (!getNamespaceTableNames(namespace).isEmpty()) {
+      throw new IllegalArgumentException(
+          "Namespace is not empty: " + namespace + ", " + getNamespaceTableNames(namespace));
+    }
+
+    try {
+      admin.dropNamespace(namespace);
+    } catch (ExecutionException e) {
+      throw new ExecutionException("Dropping the namespace failed: " + namespace, e);
+    }
+  }
+
+  @Override
+  public void truncateTable(String namespace, String table) throws ExecutionException {
+    if (!tableExists(namespace, table)) {
+      throw new IllegalArgumentException("Table does not exist: " + namespace + "." + table);
```

Comment:
```suggestion
      throw new IllegalArgumentException("Table does not exist: " + ScalarDbUtils.getFullTableName(namespace, table));
```
Likewise

Reply:
Fixed in fcb233812ead7d8c6a68a422a7467f70ba10c733. Thanks.